src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -2226,7 +2226,10 @@ private void ValidateStatementRange(StatementSyntax firstStatement, StatementSyn
                 throw new ArgumentException("statements not within tree");
             }
 
-            if (firstStatement.Parent == null || firstStatement.Parent != lastStatement.Parent)
+            if ((firstStatement.Parent == null || firstStatement.Parent != lastStatement.Parent) &&
+                // Special case for global statements, since they have no common parent that is a statement syntax
+                firstStatement.Parent is not GlobalStatementSyntax &&
+                lastStatement.Parent is not GlobalStatementSyntax)
```

Comment:
I think, if both parents are GlobalStatementSyntax, we should check the next level parents and make sure they are equal.